src/libcryptobox/cryptobox.c

```diff
@@ -360,9 +357,19 @@ void rspamd_cryptobox_keypair(rspamd_pk_t pk, rspamd_sk_t sk,
 		ec_pub = EC_POINT_new(group);
 		g_assert(ec_pub != NULL);
 
+		unsigned char *buf = NULL;
+		EVP_PKEY_get_raw_private_key(pkey, buf, &len);
```

Comment:
EVP_PKEY_get_raw_private_key() fills the buffer provided by priv with raw private key data. The size of the priv buffer should be in *len on entry to the function, and on exit *len is updated with the number of bytes actually written. If the buffer priv is NULL then *len is populated with the number of bytes required to hold the key. The calling application is responsible for ensuring that the buffer is large enough to receive the private key data. This function only works for algorithms that support raw private keys. Currently this is: EVP_PKEY_HMAC, EVP_PKEY_POLY1305, EVP_PKEY_SIPHASH, EVP_PKEY_X25519, EVP_PKEY_ED25519, EVP_PKEY_X448 or EVP_PKEY_ED448.

Hence, this will not allocate `buf` at all, it will only populate `len` required for it. We definitely need tests for these changes, as OpenSSL API is way too confusing and dangerous.